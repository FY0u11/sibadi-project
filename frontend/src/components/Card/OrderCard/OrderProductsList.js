import React from 'react'

const OrderProductsList = ({ orderProducts }) => {
  return (
    <div>
      <h5>Product list</h5>
      <ol>
        {orderProducts.map(product => (
          <li key={product.id}>
            {product.name} <span className="small text-muted">{product.price} $</span>
          </li>
        ))}
      </ol>
    </div>
  )
}

export default OrderProductsList
