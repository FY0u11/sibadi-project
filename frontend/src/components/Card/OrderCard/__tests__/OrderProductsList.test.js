import React from 'react'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import OrderProductsList from '../OrderProductsList'

afterEach(cleanup)

describe('Testing OrderProductsList component', () => {
  it('It should be properly rendered', () => {
    const orderProducts = [
      { id: 2, name: 'product 1', price: 100 },
      { id: 3, name: 'product 2', price: 250 }
    ]
    const { asFragment } = render(<OrderProductsList orderProducts={orderProducts} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
