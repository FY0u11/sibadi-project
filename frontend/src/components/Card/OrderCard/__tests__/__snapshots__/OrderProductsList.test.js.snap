// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing OrderProductsList component It should be properly rendered 1`] = `
<DocumentFragment>
  <div>
    <h5>
      Product list
    </h5>
    <ol>
      <li>
        product 1 
        <span
          class="small text-muted"
        >
          100 $
        </span>
      </li>
      <li>
        product 2 
        <span
          class="small text-muted"
        >
          250 $
        </span>
      </li>
    </ol>
  </div>
</DocumentFragment>
`;
